modules/asset-manager-impl/src/main/java/org/opencastproject/assetmanager/impl/AssetManagerWithSecurity.java

```diff
@@ -181,15 +174,16 @@ private Predicate restrictToUsersOrganization() {
   }
 
   /** Check authorization based on the given predicate. */
-  private boolean isAuthorized(Predicate p) {
+  private boolean isAuthorized(final String mediaPackageId, final String action) {
     switch (isAdmin()) {
       case GLOBAL:
         return true;
       case ORGANIZATION:
         return true;
       default:
-        final AQueryBuilder q = delegate.createQuery();
-        return !q.select().where(p).run().getRecords().isEmpty();
+        return !delegate.createQuery().select()
+                .where(mkAuthPredicate(mediaPackageId, action))
+                .run().getRecords().isEmpty();
```

Comment:
`mkAuthPredicate(…)` was called every time `isAuthorized(…)` was called through the result was not necessarily used. It makes more sense to call this in here and only if it is needed.